pycqed/measurement/waveform_control/segment.py

```diff
@@ -810,15 +829,96 @@ def _test_overlap(self, allow_overlap=False, tol=1e-12):
                 el_new_start = el_list[i + 1][0]
 
                 if (el_prev_end - el_new_start) > tol :
-                    print(el_prev_end, el_new_start)
-                    msg = f'{prev_el} (ends at {el_prev_end*1e6:.4f}us) and ' \
-                    f'{el_list[i + 1][2]} (' \
-                        f'starts at {el_new_start*1e6:.4f}us) overlap ' \
-                          f'on {awg}'
-                    if allow_overlap:
-                        log.warning(msg)
+                    if track_and_ignore:
+                        # add set of two overlapping elements to list
```

Comment:
Have you tested whether your code works correctly in case of more than two overlapping elements? Looking at the criterion `(el_prev_end - el_new_start) > tol`, I wonder whether there are cases where an overlapped might stay undetected (e.g., a long element starting first plus two short non-overlapping elements that both overlap with the long element).

Reply:
Indeed this is a very good comment. When writing this I wanted to reuse the code that we already use for testing whether two elements overlap. If the working assumption is, that elements do not overlap, this criterion is sufficient for detecting *one* overlapping element. As you noticed, it does not work if the next element overlaps with the second to last one (or elements before that).  

I do see to solutions to this problem:
1.) when finding an overlapping element, test whether this element also overlaps with any of the following elements. 
2.) use an entirely different approach for testing for overlaps, trying to test overlaps of all elements with any of the other elements. 

While approach 2.) is more elegant in the sense that it is more general (less if statements), it might be computationally more intense (scales with num_el**2). Especially since the general case will still be to not have overlapping elements, such a test is not required in most instances. 

I will go ahead with testing 1.) (as it is also easier to implement). Curious to hear your opinion @chellings

Reply:
I pushed my suggested solution (I tested it for your suggested case and also the other cases I tested before).

Reply:
Fine by me, the suggested solution sounds reasonable. I can have a final quick look at the code ASAP.